hpx/runtime/actions/basic_action.hpp

```diff
@@ -290,23 +308,18 @@ namespace hpx { namespace actions
         // case no continuation has been supplied.
         template <typename ...Ts>
         static threads::thread_function_type
-        construct_thread_function(naming::id_type const& target,
+        construct_thread_function(naming::id_type target,
             naming::address::address_type lva,
             naming::address::component_type comptype, Ts&&... vs)
         {
             if (target &&
                 target.get_management_type() == naming::id_type::unmanaged)
-            {
-                return traits::action_decorate_function<Derived>::call(lva,
-                    util::one_shot(util::bind(
-                        typename Derived::thread_function(),
-                        lva, comptype, std::forward<Ts>(vs)...)));
-            }
+                target = {};
 
+            using thread_function = detail::thread_function<Derived>;
             return traits::action_decorate_function<Derived>::call(lva,
-                util::one_shot(util::bind(
-                    typename Derived::thread_function(target),
-                    lva, comptype, std::forward<Ts>(vs)...)));
+                thread_function(target, lva, comptype,
```

Comment:
Should `target` be moved here?

Reply:
It's now a local copy so it should at least not hurt moving it.

Reply:
Moving it would safe an atomic increment. I'd suggest doing it.